qgreenland_server.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+"""
+/***************************************************************************
+ QGreenlandDownload
+                                 A QGIS plugin
+ Download QGreenland dataset
+ Generated by Plugin Builder: http://g-sherman.github.io/Qgis-Plugin-Builder/
+                             -------------------
+        begin                : 2021-08-17
+        git sha              : $Format:%H$
+        copyright            : (C) 2021 by Matteo Ghetta (Faunalia)
+        email                : matteo.ghetta@faunalia.eu
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+
+import os
+import json
+
+from qgis.PyQt import uic, QtWidgets
+
+from qgis.core import(
+    QgsSettings
+)
+
+# This loads your .ui file so that PyQt can populate your plugin with the elements from Qt Designer
+FORM_CLASS, _ = uic.loadUiType(os.path.join(
+    os.path.dirname(__file__), 'ui','qgreenland_server_dialog.ui'))
+
+
+
+class QGreenlandServer(QtWidgets.QDialog, FORM_CLASS):
+    def __init__(self, parent=None):
+        """Constructor."""
+        super(QGreenlandServer, self).__init__(parent)
+        self.setupUi(self)
+
+        self.settings = QgsSettings()
```

Comment:
I'm still trying to learn how this works so thanks in advance for your help understanding :) Does `QgsSettings()` return the same settings object to `QGreenlandServer` as it does for `QGreenlandDownload`, or does it create a new instance?

Reply:
basically with `QgsSettings` you are dealing with the settings of the profile folder of QGIS. So even if it is a new instance in another file (and class) you have access to the `QGIS.ini` file variables. Just have a look in the `QGIS` folder within the user profile and open the `QGIS.ini` file and search for QGreenland :)